docs/src/main/asciidoc/class-loading-reference.adoc

```diff
@@ -191,3 +172,39 @@ file). Simply add an `excludedArtifacts` section as shown below:
 
 This should only be done if the extension depends on a newer version of these artifacts. If the extension does not bring
 in a replacement artifact as a dependency then classes the application needs might end up missing.
+
+== Configuring Class Loading
+
+It is possible to configure some aspects of class loading in dev and test mode. This can be done using `application.properties`.
+Note that class loading config is different to normal config, in that it does not use the standard Quarkus config mechanisms
+(as it is needed too early), so only supports `application.properties`. The following options are supported.
+
+
+`quarkus.class-loading.parent-first-artifacts`::
+A comma separated list of artifacts (in the groupId:artifactId format) that will be loaded parent first. Note that if you
+make a library parent first all its dependencies should generally also be parent first.
+
+`quarkus.class-loading.reloadable-artifacts`::
+A comma separated list of artifacts (in the groupId:artifactId format) that will be loaded in the runtime ClassLoader. This
+means they will be recreated on every hot reload. This can be used to work around issues with 3rd party libraries that
+save static state. Note that if you use this any library that depends on the listed libraries will also need to be reloadable.
```

Comment:
You could use the generated `quarkus-class-loading-configuration-class-loading-config.adoc` file.